vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3033,8 +3033,7 @@ public void setDetailsVisible(T item, boolean visible) {
     public void setDetailsVisibleOnClick(boolean detailsVisibleOnClick) {
         if (this.detailsVisibleOnClick != detailsVisibleOnClick) {
             this.detailsVisibleOnClick = detailsVisibleOnClick;
```

Comment:
The internal `detailsVisibleOnClick` property can assumably be removed also (to avoid double bookkeeping). The `isDetailsVisibleOnClick()` could be modified to read the element property `!getElement().getProperty("__disallowDetailsOnClick", false)`